slurm-on-sles12-hpc/azuredeploy.json

```diff
@@ -0,0 +1,559 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "dnsName": {
```

Comment:
dnsname and storageaccounts cause a bunch of template failures due to strict requirements, I suggest creating variables for these and generating these using the uniqueString() function. See usage here https://github.com/Azure/azure-quickstart-templates/blob/bfb0c40965969a4884ccee65ff75c7ee13e3195b/301-2fe-lb80-rdp-1be-nsg-rdp/azuredeploy.json#L32